### PR TITLE
K closest points

### LINKED ISSUE #12: Challenge: K nearest points
Develop a python function to find a given number of closest points to the point (0,0) in a list of points

### PR DESCRIPTION
**Pull request by Fahim-Ahmad**
- [x] PR's title includes the challenge's name and language
- [x] PR has short and clear description of the challenge
- [x] PR has appropriate labels and milestones for easy identification
- [x] PR it is assigned to the owner
- [x] reviewers are assigned
- [x] the PR contributes only one focused change
- [x] the branch is up to date with `main`/`master`
- [x] the code works when pulled and run locally
- [x] all conflicts are resolved (if any)
- [x] It is linked to an issue in the appropriate column of the project board
- [ ] feedback is addressed (if any, and if it is appropriate feedback.)

## README Documentation

- [x] The solution is documented in `/src/README.md`
- [x] The markdown source is formatted
- [x] Spelling and grammar is correct in all text
- [ ] The markdown looks correct when you preview the file
- [x] All links and images work
- [ ] The README documents the solution's behavior, strategy(ies) and implementation(s)
- [x] There are use cases in code blocks to illustrate the function's behavior

## Python Files

- [ ] There is a module header
- [ ] There is a module docstring
- [x] File names are in snake_case
- [x] Test files are named `test_<module_name>.py`

## Function Docstring

- [ ] Behavior description
- [ ] Parameter description
- [ ] Return value description
- [ ] Include any assumptions
- [ ] Include 3 or more (passing!) doctests
- [ ] Include 1-2 use cases (if necessary)

## Function Implementation

- [x] The solution is not is not copied from [djeada](https://github.com/djeada/Algorithms-And-Data-Structures)
- [x] The code is auto-formatted
- [x] The code has no (reasonable) linting mistakes
- [x] Variables are named with snake_case
- [x] The function has a clear and helpful name
- [x] The file's name matches the function name
- [x] The code follows the strategy as simply as possible
- [x] Variable names are clear and helpful
- [x] Comments explain the strategy (if necessary)
- [ ] There are type annotations
- [x] (challenge) The code includes defensive assertions

## Unit Test Suite

- [x] The test class has a helpful name in PascalCase
- [x] The test class has a docstring
- Each unit test has
  - [x] A helpful name
  - [x] A clear docstring
  - [x] Only one assertion
- [x] All tests pass
- [x] There are tests for defensive assertions
- [x] There are tests for boundary cases
- [x] Test suite includes black-box and glass-box tests

### CODE REVIEW

File: src/brain_teasers/python/k_closest_points/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# K Closest Points to (0, 0)
+
+This python function finds the K closest points to (0, 0) in a given list of points.
+
+```
```
Comment (mohammedalasly) [inline src/brain_teasers/python/k_closest_points/README.md]: This looks nice, and you can add `python` after the backtick  it will make the documented code more readable
```python
# your code
```

File: src/brain_teasers/python/k_closest_points/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+k_closest_points([(-3, -4), (0, 0), (1, 1), (2, 2), (-1, -1), (-5, -6)], 3)
+output: [(0, 0), (1, 1), (-1, -1)]
+
+k_closest_points([(1, 2), (3, 4), (-1, 0), (5, 6), (0, 0), (8, 8)], 2)
+output: [(0, 0), (-1, 0)]
+
+k_closest_points([(-3, -4), (0, 0), (1, 1), (0, 0), (0, 0), (-5, -6)], 3)
+output: [(0, 0), (0, 0), (0, 0)]
+
+```
```
Comment (mohammedalasly) [inline src/brain_teasers/python/k_closest_points/README.md]: you aslo need to add the command line that tests the code, which will improve the README file documentation
```pash
python -m unittest discover -v
```

Comment (mohammedalasly) [inline src/brain_teasers/python/k_closest_points/README.md]: Always include a conclusion in your **README** file that summarizes what the module does and its approach

**Review by mohammedalasly — CHANGES_REQUESTED** [review]

File: src/brain_teasers/python/k_closest_points/src/k_closest_points.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+def k_closest_points(points, k):
```
Comment (mohammedalasly) [inline src/brain_teasers/python/k_closest_points/src/k_closest_points.py]: Your code looks nice, short, and easy to read, and it passed the test when running it locally.
Just for you to fulfill the checklist requirements, you need to add a "Module Header" on top 
'''Module header: K_closest_points
  _short explanation about it_
'''

File: src/brain_teasers/python/k_closest_points/src/k_closest_points.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+    def distance_function(point):
```
Comment (mohammedalasly) [inline src/brain_teasers/python/k_closest_points/src/k_closest_points.py:19]: You can also use `type annotations` instead of regular parameters,

File: src/brain_teasers/python/k_closest_points/src/k_closest_points.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+        return ((point[0]-0)**2 + (point[1]-0)**2)**0.5
```
Comment (mohammedalasly) [inline src/brain_teasers/python/k_closest_points/src/k_closest_points.py:26]: Use function docstring to add `Behavior description`  and `Parameter description`

**Review by mohammedalasly — CHANGES_REQUESTED** [review]

File: src/brain_teasers/python/k_closest_points/README.md (hunk 1 above)
Comment (Fahim-Ahmad) [reply src/brain_teasers/python/k_closest_points/README.md]: Added : )

File: src/brain_teasers/python/k_closest_points/README.md (hunk 2 above)
Comment (Fahim-Ahmad) [reply src/brain_teasers/python/k_closest_points/README.md]: Thanks. The readme file is updated with the instruction for testing code implementation.

Comment (Fahim-Ahmad) [reply src/brain_teasers/python/k_closest_points/README.md]: Thanks. I added a summary.

File: src/brain_teasers/python/k_closest_points/src/k_closest_points.py (hunk 3 above)
Comment (Fahim-Ahmad) [reply src/brain_teasers/python/k_closest_points/src/k_closest_points.py]: Thanks for noticing it. I added a module header.

File: src/brain_teasers/python/k_closest_points/src/k_closest_points.py (hunk 5 above)
Comment (Fahim-Ahmad) [reply src/brain_teasers/python/k_closest_points/src/k_closest_points.py:26]: Excellent. Thanks for your attention to details.
the docstring is now added.

**Review by mohammedalasly — APPROVED** [review]
Looks very nice, great job!!
